src/Identifier/Ldap/ExtensionAdapter.php

```diff
@@ -161,6 +161,23 @@ public function unbind()
         $this->_connection = null;
     }
 
+    /**
+     * Search the LDAP directory
+     *
+     * @param string $baseDN Base DN for the directory
+     * @param string $filter search filter
+     * @return array
+     */
+    public function search($baseDN, $filter)
+    {
+        $this->_setErrorHandler();
+        $result = ldap_search($this->_connection, $baseDN, $filter);
+        $entries = ldap_get_entries($this->_connection, $result);
```

Comment:
If these lines result in an error being thrown you won't restore the error handler.